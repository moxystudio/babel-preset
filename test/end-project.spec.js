'use strict';

const serializerPath = require('jest-serializer-path');
const preset = require('../end-project');

expect.addSnapshotSerializer(serializerPath);

it('should have the right config with the default options', () => {
    expect(preset()).toMatchSnapshot();
});

describe('modules', () => {
    it('should set modules to false if options.modules is disabled', () => {
        expect(preset(null, { modules: false })).toMatchSnapshot();
    });

    it('should set modules to false by default if BABEL_ENV is es', () => {
        process.env.BABEL_ENV = 'es';

        try {
            expect(preset(null)).toMatchSnapshot();
        } finally {
            delete process.env.BABEL_ENV;
        }
    });
});

describe('react', () => {
    it('should enable react plugins if options.react is enabled and enable development goodies', () => {
        expect(preset(null, { react: true })).toMatchSnapshot();
    });

    it('should enable react optimizations if options.env is production', () => {
        expect(preset(null, { env: 'production', react: true })).toMatchSnapshot();
    });
});

describe('lodash', () => {
    it('should disable the lodash plugin if options.lodash is disabled', () => {
        expect(preset(null, { lodash: false })).toMatchSnapshot();
    });

    it('should pass any options to the lodash plugin', () => {
        expect(preset(null, { lodash: { id: 'recompose' } })).toMatchSnapshot();
    });
});

describe('targets', () => {
    it('should return a default config that target browsers and node', () => {
        expect(preset()).toMatchSnapshot();
    });

    it('should return a config for node if options.targets is set to node', () => {
        expect(preset(null, { targets: ['node'] })).toMatchSnapshot();
    });

    it('should return a config for browsers if options.targets is set to browsers', () => {
        expect(preset(null, { targets: ['browsers'] })).toMatchSnapshot();
    });

    it('should allow overriding options.targets with an object', () => {
        expect(preset(null, { targets: { node: 'current' } })).toMatchSnapshot();
    });
});

describe('env', () => {
    it('should take into consideration process.env.NODE_ENV', () => {
        process.env.NODE_ENV = 'development';
        expect(preset(null, { react: true })).toMatchSnapshot();
        process.env.NODE_ENV = 'production';
        expect(preset(null, { react: true })).toMatchSnapshot();
    });
});

describe('dynamicImport', () => {
    it('should take into consideration options.dynamicImport', () => {
        expect(preset(null, { dynamicImport: false })).toMatchSnapshot();
    });
});
