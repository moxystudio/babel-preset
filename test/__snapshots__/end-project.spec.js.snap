// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamicImport should take into consideration options.dynamicImport 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`env should take into consideration process.env.NODE_ENV 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx-source/lib/index.js",
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx-self/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`env should take into consideration process.env.NODE_ENV 2`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-remove-prop-types/lib/index.js",
      Object {
        "removeImport": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`functional class-properties should handle correctly 1`] = `
"class Bork {
  constructor() {
    this.x = 'bar';
    this.y = void 0;
  }

}

Bork.a = 'foo';
Bork.b = void 0;"
`;

exports[`functional nullish-coalescing-operator should handle correctly 1`] = `
"var _object$foo;

const foo = (_object$foo = object.foo) != null ? _object$foo : 'bar';"
`;

exports[`functional object-rest-spread should handle correctly 1`] = `
"var z = {
  x,
  ...y
};"
`;

exports[`functional object-rest-spread should handle correctly for target with support 1`] = `
"var z = {
  x,
  ...y
};"
`;

exports[`functional optional-chaining should handle correctly 1`] = `
"var _obj$foo, _obj$foo$bar;

const obj = {
  foo: {
    bar: {
      baz: 42
    }
  }
};
const baz = obj == null ? void 0 : (_obj$foo = obj.foo) == null ? void 0 : (_obj$foo$bar = _obj$foo.bar) == null ? void 0 : _obj$foo$bar.baz;"
`;

exports[`lodash should disable the lodash plugin if options.lodash is disabled 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`lodash should pass any options to the lodash plugin 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
          "recompose",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`loose should disable loose if options.loose is disabled 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`modules should set modules to false by default if BABEL_ENV is es 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": false,
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`modules should set modules to false if options.modules is disabled 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": false,
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`react should enable react optimizations if options.env is production 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-transform-react-remove-prop-types/lib/index.js",
      Object {
        "removeImport": true,
      },
    ],
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`react should enable react plugins if options.react is enabled and enable development goodies 1`] = `
Object {
  "plugins": Array [
    "<PROJECT_ROOT>/node_modules/@babel/plugin-syntax-jsx/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-jsx/lib/index.js",
      Object {
        "useBuiltIns": true,
      },
    ],
    "<PROJECT_ROOT>/node_modules/@babel/plugin-transform-react-display-name/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`should have the right config with the default options 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`targets should allow overriding options.targets with an object 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "node": "current",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`targets should return a config for browsers if options.targets is set to browsers 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`targets should return a config for node if options.targets is set to node 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;

exports[`targets should return a default config that target browsers and node 1`] = `
Object {
  "plugins": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
      Object {
        "loose": true,
      },
    ],
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "absoluteRuntime": "<PROJECT_ROOT>/node_modules/@babel/runtime",
        "corejs": false,
        "helpers": false,
        "regenerator": true,
        "useESModules": false,
      },
    ],
    "<PROJECT_ROOT>/node_modules/babel-plugin-dynamic-import-node/lib/index.js",
    Array [
      "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
      Object {
        "id": Array [
          "lodash",
          "lodash-es",
          "lodash-compat",
          "lodash/fp",
        ],
      },
    ],
  ],
  "presets": Array [
    Array [
      "<PROJECT_ROOT>/node_modules/@babel/preset-env/lib/index.js",
      Object {
        "corejs": 3,
        "loose": true,
        "modules": "commonjs",
        "shippedProposals": true,
        "targets": Object {
          "browsers": Array [
            "extends browserslist-config-google/modern",
          ],
          "node": "8.9",
        },
        "useBuiltIns": "entry",
      },
    ],
  ],
  "sourceType": "unambiguous",
}
`;
